### ISSUE TITLE
RetryPolicy could be a BoundedLattice

### ISSUE THREAD

**Issue by LukaJCB** [cb372/cats-retry#38]
Right now for `combine` we're using `max` as the operation and `Duration.Zero` as the identity element. We could also introduce an operation using `min` and `Duration.Inf` as the identity element. 
This would mean we have `max` as `join` and `min` as `meet`. We might even be able to go further and define a `BoundedDistributiveLattice`, though off-hand I'm not sure if `min` and `max` distribute.

Unfortunately getting these typeclass instances would mean depending on something like `typelevel/algebra`, which I'm not sure we want to do, but it might make sense to just define `join` and `meet` on the type itself. WDYT? :)

**Comment by cb372** [comment]
Interesting idea! Do you have an example of how this would be useful in the context of composing retry policies? So far I've always found composition using `max` to make sense but I'm interested in hearing about other use cases I haven't encountered.

As you say, I'd be wary of adding a dependency on algebra just for this, unless it could be shown to be really useful.

**Comment by LukaJCB** [comment]
To be clear this is what I have in mind for `meet`:

- If either of the policies wants to try again, the two policies combined will try again
- If both policies want to delay and retry, the shorter of the two delays is chosen

So a simple example is when you have a an exponential backoff, but you don't want to go further than a certain threshold, so you can simply say:

```scala
constantDelay(5.seconds) meet exponentialBackoff(10.millis)
```

That said, even if this is really useful I don't think we should add the dependency given that there aren't a ton of combinators on top of `BoundedLattice`, it's just a nice way to guarantee certain properties for users of this library IMO. :)

**Comment by cb372** [comment]
Fixed by #44
